packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.tsx

```diff
@@ -133,21 +131,50 @@ export const CharacterCounter = React.forwardRef<
       }
       return `${maxCharacters} ${i18n.characterCounter.charactersAllowed}`;
     }
-  }
+  }, [characterLimit, inputLength, i18n.characterCounter, maxCharacters]);
+
+  // Sets the screen reader message to announce the character counter.
+  const [screenReaderMessage, setScreenReaderMessage] = React.useState('');
+
+  const debouncedSetScreenReaderMessage = React.useMemo(
+    () =>
+      debounce((statusMessage: string) => {
+        setScreenReaderMessage(statusMessage);
+      }, 500),
+    []
+  );
+
+  React.useEffect(() => {
+    setTimeout(() => {
```

Comment:
What does this additional 1 sec timeout do on top of the 500ms debounce?

Reply:
It helped with the repetition from the screen reader I was experiencing from each key press.